src/server/scripts/Northrend/Gundrak/boss_moorabi.cpp

```diff
@@ -125,41 +138,49 @@ class boss_moorabi : public CreatureScript
                 }
             }
 
-            void ExecuteEvent(uint32 eventId) override
+            void UpdateAI(uint32 diff) override
             {
-                switch (eventId)
+                if (!me->IsInCombat())
+                    scheduler.Update(diff);
```

Comment:
Event handled by TaskScheduler is supposed to happen only out of combat, so i think the only other way to handle this is to put `events.Update(diff);` above everything (no victim out of combat).